ComplianceAsCode/content
# Add support for XCCDF variables into sshd_lineinfile template

#### Description:

- modify the sshd_lineinfile template and related code so that the value of sshd parameter can be specified through an xccdf variable
    - create new macros for lineinfile states which use XCCDF variable
    - modify the OVAL, Bash and Ansible code used in the template
    - modify all templated test scenarios so that they use the variable
    - add a new required parameter datatype. It allows to create correct test scenarios. Note that this could be probably omitted, but I think that having proper support for data types in this template could allow us greater flexibility in the future, for example checking if some value is greater or equal.
- modify all rules templated with sshd_lineinfile to include the datatype parameter
- template the rule sshd_set_keepalive as a working POC
- update the sshd_lineinfile template documentation

#### Rationale:

The sshd_lineinfile is very robust and it has support for distributed sshd configuration.
There are rules regarding sshd configuration which utilize XCCDF variables.
But these rules are not using the sshd_lineinfile template and therefore they don't respect presence of distributed sshd configuration. This creates problems because if two categories of rules mentioned above are combined, then remediations can end up in different files; some in the main sshd_config and some in a file within the sshd_config.d directory. This is not desired.

Please note that I did not template all relevant rules, I modified only one to keep the PR size down. The rest will follow up in a separate PR.

#### Review Hints:

- running automatus in template mode should be sufficient
